src/jdk.attach/linux/native/libattach/VirtualMachineImpl.c

```diff
@@ -194,7 +194,10 @@ JNIEXPORT void JNICALL Java_sun_tools_attach_VirtualMachineImpl_close
 {
     int res;
     shutdown(fd, SHUT_RDWR);
-    RESTARTABLE(close(fd), res);
+    res = close(fd);
+    if (res == -1) {
+        JNU_ThrowIOExceptionWithLastError(env, "close");
+    }
```

Comment:
I assume it would be better to not stop when errno is EINTR. If there is a profiler or some other tool firing signals at threads then there isn't anything that can be done here.

Reply:
Should I just ignore the result / return code of close and avoid throwing an exception completely ?

Reply:
I would just ignore any error on close in this case and treat it as a best-effort attempt at closing the fd. The close could be successful even if an error is reported; conversely if a signal-based profiler is used (whether they use SA_RESTART or not) we don't want the profiler's actions to cause application/library exceptions.